db/dohast.c

```diff
@@ -337,6 +342,13 @@ static void node_free(dohsql_node_t **pnode, sqlite3 *db)
         node_free(&node->nodes[i], db);
     }
 
+    /* params */
+    if ((*pnode)->params) {
+        if ((*pnode)->params->params)
+            free((*pnode)->params->params);
```

Comment:
You don't need to check for NULL. `free(NULL)` is a no-op.

Reply:
I didn't want to make assumptions about our code that overrides free, but sure.